lib/rules/no-useless-escape.js

```diff
@@ -182,6 +150,131 @@ module.exports = {
+
+                        // It's not an escaped character.
+                        return;
+                    }
+
+                    const escapedChar = characterNode.raw.slice(1);
+
+                    if (escapedChar !== String.fromCodePoint(characterNode.value)) {
+
+                        // It's a valid escape.
+                        return;
+                    }
+                    let allowedEscapes;
+
+                    if (characterClassStack.length) {
+                        allowedEscapes = unicodeSets ? REGEX_CLASSSET_CHARACTER_ESCAPES : REGEX_GENERAL_ESCAPES;
+                    } else {
+                        allowedEscapes = REGEX_NON_CHARCLASS_ESCAPES;
+                    }
+                    if (allowedEscapes.has(escapedChar)) {
+                        return;
+                    }
+
+                    const reportedIndex = characterNode.start + 1;
+
+                    if (characterClassStack.length) {
+                        if (!unicodeSets) {
+                            const characterClassNode = characterClassStack[0];
+
+                            if (characterClassNode.type === "CharacterClass") {
+                                if (escapedChar === "-") {
+
+                                    /*
+                                     * The '-' character is a special case, because it's only valid to escape it if it's in a character
+                                     * class, and is not at either edge of the character class. To account for this, don't consider '-'
+                                     * characters to be valid in general, and filter out '-' characters that appear in the middle of a
+                                     * character class.
+                                     */
+                                    if (characterClassNode.start + 1 !== characterNode.start && characterNode.end !== characterClassNode.end - 1) {
+
+                                        return;
+                                    }
+                                }
+                                if (escapedChar === "^") {
+
+                                    /*
+                                     * The '^' character is also a special case; it must always be escaped outside of character classes, but
+                                     * it only needs to be escaped in character classes if it's at the beginning of the character class. To
+                                     * account for this, consider it to be a valid escape character outside of character classes, and filter
+                                     * out '^' characters that appear at the start of a character class.
+                                     */
+                                    if (characterClassNode.start + 1 === characterNode.start) {
+
+                                        return;
+                                    }
+                                }
```

Comment:
This should apply in unicodeSets mode, too.

```js
/* eslint no-useless-escape: "error" */

/[\^]/v; // false positive

```

Reply:
Thank you. I overlooked it. I will fix it.

Reply:
I fixed it.
https://github.com/eslint/eslint/pull/17420/commits/ef10e1ac6107afa65fe3b8697048d99b79b6ba14